### PR TITLE
Fix/100165 rsvp change status capacity

### PR DESCRIPTION
**Pull request by jesseeproductions**
_Ref:_ [C#100165](https://central.tri.be/issues/100165)

### CODE REVIEW

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | There must be a single space after the first parenthesis | [Tickets_View.php:194](https://github.com/moderntribe/event-tickets/blob/58969884ebc3a5874c8cbcad0d7b8c5be6b7efeb/src/Tribe/Tickets_View.php#L194)
:x: | There must be a single space before the last parenthesis | [Tickets_View.php:194](https://github.com/moderntribe/event-tickets/blob/58969884ebc3a5874c8cbcad0d7b8c5be6b7efeb/src/Tribe/Tickets_View.php#L194)
:x: | There must be a single space before the last parenthesis | [Tickets_View.php:194](https://github.com/moderntribe/event-tickets/blob/58969884ebc3a5874c8cbcad0d7b8c5be6b7efeb/src/Tribe/Tickets_View.php#L194)
:x: | Missing space after array opener. | [Tickets_View.php:194](https://github.com/moderntribe/event-tickets/blob/58969884ebc3a5874c8cbcad0d7b8c5be6b7efeb/src/Tribe/Tickets_View.php#L194)
:x: | There must be a single space after the first parenthesis | [Tickets_View.php:194](https://github.com/moderntribe/event-tickets/blob/58969884ebc3a5874c8cbcad0d7b8c5be6b7efeb/src/Tribe/Tickets_View.php#L194)
:x: | There must be a single space after the first parenthesis | [Tickets_View.php:194](https://github.com/moderntribe/event-tickets/blob/58969884ebc3a5874c8cbcad0d7b8c5be6b7efeb/src/Tribe/Tickets_View.php#L194)
:x: | There must be a single space before the last parenthesis | [Tickets_View.php:194](https://github.com/moderntribe/event-tickets/blob/58969884ebc3a5874c8cbcad0d7b8c5be6b7efeb/src/Tribe/Tickets_View.php#L194)
:x: | Missing space before array closer. | [Tickets_View.php:194](https://github.com/moderntribe/event-tickets/blob/58969884ebc3a5874c8cbcad0d7b8c5be6b7efeb/src/Tribe/Tickets_View.php#L194)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | There must be a single space after the first parenthesis | [Tickets_View.php:194](https://github.com/moderntribe/event-tickets/blob/8546d2dc74b4d68167463eb97e5867080b36cfa7/src/Tribe/Tickets_View.php#L194)
:x: | There must be a single space before the last parenthesis | [Tickets_View.php:194](https://github.com/moderntribe/event-tickets/blob/8546d2dc74b4d68167463eb97e5867080b36cfa7/src/Tribe/Tickets_View.php#L194)
:x: | There must be a single space before the last parenthesis | [Tickets_View.php:194](https://github.com/moderntribe/event-tickets/blob/8546d2dc74b4d68167463eb97e5867080b36cfa7/src/Tribe/Tickets_View.php#L194)
:x: | Missing space after array opener. | [Tickets_View.php:194](https://github.com/moderntribe/event-tickets/blob/8546d2dc74b4d68167463eb97e5867080b36cfa7/src/Tribe/Tickets_View.php#L194)
:x: | There must be a single space after the first parenthesis | [Tickets_View.php:194](https://github.com/moderntribe/event-tickets/blob/8546d2dc74b4d68167463eb97e5867080b36cfa7/src/Tribe/Tickets_View.php#L194)
:x: | There must be a single space after the first parenthesis | [Tickets_View.php:194](https://github.com/moderntribe/event-tickets/blob/8546d2dc74b4d68167463eb97e5867080b36cfa7/src/Tribe/Tickets_View.php#L194)
:x: | There must be a single space before the last parenthesis | [Tickets_View.php:194](https://github.com/moderntribe/event-tickets/blob/8546d2dc74b4d68167463eb97e5867080b36cfa7/src/Tribe/Tickets_View.php#L194)
:x: | Missing space before array closer. | [Tickets_View.php:194](https://github.com/moderntribe/event-tickets/blob/8546d2dc74b4d68167463eb97e5867080b36cfa7/src/Tribe/Tickets_View.php#L194)

> via jenkins/codesniffer-phpcs

File: src/Tribe/RSVP.php [hunk 1]
```diff
@@ -402,6 +402,16 @@ public function update_attendee_data( $data, $order_id, $event_id ) {
 
 		$product_id  = $attendee['product_id'];
 
+		//check if changing status will cause rsvp to go over capacity
+		$previous_order_status = get_post_meta( $order_id, self::ATTENDEE_RSVP_KEY, true );
+		if ( 'yes' === $attendee_order_status && 'no' === $previous_order_status ) {
```
Comment (bordoni) [inline src/Tribe/RSVP.php]: use `tribe_is_truthy()`

File: src/Tribe/Tickets_View.php [hunk 2]
```diff
@@ -170,6 +185,14 @@ public function update_tickets() {
 		$post_id = get_the_ID();
 		$attendees = ! empty( $_POST['attendee'] ) ? $_POST['attendee'] : array();
 
+		/**
+		 * Sort list to handle all not attending first
+		 *
+		 * switch to the following once WordPress 4.7 is minimum supported:
+		 * $attendees = wp_list_sort( $attendees, 'order_status', 'ASC', true );
```
Comment (bordoni) [inline src/Tribe/Tickets_View.php]: I would check if `wp_list_sort` exists, use it, otherwise fall back to what you did, and add a `@todo` for that.

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Comment refers to a TODO task | [Tickets_View.php:191](https://github.com/moderntribe/event-tickets/blob/5f0285112f87619a7fe0037f6f1aa43335e13255/src/Tribe/Tickets_View.php#L191)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Comment refers to a TODO task | [Tickets_View.php:191](https://github.com/moderntribe/event-tickets/blob/a4552d3024c50cbdc23fcf518d69c7b5455e6c74/src/Tribe/Tickets_View.php#L191)

> via jenkins/codesniffer-phpcs

**Comment by ggwicz** [comment]
There are some [failing tests on the travis-ci build](https://travis-ci.org/moderntribe/event-tickets/jobs/352317847) — not sure if they're related to this PR (doesn't seem like it), but wanted to flag.

Also, f some reason GitHub isn't letting me dismiss @bordoni's review.

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Comment refers to a TODO task | [Tickets_View.php:191](https://github.com/moderntribe/event-tickets/blob/057bc629752efe3220915221006c8dfca71b47bc/src/Tribe/Tickets_View.php#L191)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Comment refers to a TODO task | [Tickets_View.php:192](https://github.com/moderntribe/event-tickets/blob/a0756edf7a20ab309ceddd43928fb55ede5af7ba/src/Tribe/Tickets_View.php#L192)

> via jenkins/codesniffer-phpcs

**Pull request merged**